File: rust/src/models/coupon_list_response.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{coupon::Coupon, pagination_response::PaginationResponse};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct CouponListResponse {
    pub data: Vec<Coupon>,

    pub pagination_meta: PaginationResponse,
}

impl CouponListResponse {
    pub fn new(data: Vec<Coupon>, pagination_meta: PaginationResponse) -> Self {
        Self {
            data,
            pagination_meta,
        }
    }
}

File: typescript/src/models/couponListResponse.ts
// this file is @generated
import { type Coupon, CouponSerializer } from "./coupon";
import {
  type PaginationResponse,
  PaginationResponseSerializer,
} from "./paginationResponse";

export interface CouponListResponse {
  data: Coupon[];

  paginationMeta: PaginationResponse;
}

export const CouponListResponseSerializer = {
  _fromJsonObject(object: any): CouponListResponse {
    return {
      data: object["data"].map((item: any) => CouponSerializer._fromJsonObject(item)),
      paginationMeta: PaginationResponseSerializer._fromJsonObject(
        object["pagination_meta"]
      ),
    };
  },

  _toJsonObject(self: CouponListResponse): any {
    return {
      data: self.data.map((item: any) => CouponSerializer._toJsonObject(item)),
      pagination_meta: PaginationResponseSerializer._toJsonObject(self.paginationMeta),
    };
  },
};
